File: unibo.basicrobot23/resources/robotNano/Motors.py
## Adattato da
## https://www.electronicshub.org/raspberry-pi-l298n-interface-tutorial-control-dc-motor-l298n-raspberry-pi/
import RPi.GPIO as GPIO
import sys
from time import sleep

GPIO.setmode(GPIO.BCM)
#MOTORE DESTRO
in1 = 3        # 9 wpi
in2 = 2        # 8 wpi
#MOTORE SINISTRO
in3 = 10       # 12 wpi
in4 = 9        # 13 wpi
## en  = 18

## GPIO.setup(en,GPIO.OUT)

GPIO.setmode(GPIO.BCM)
GPIO.setup(in1,GPIO.OUT)
GPIO.setup(in2,GPIO.OUT)
GPIO.output(in1,GPIO.LOW)
GPIO.output(in2,GPIO.LOW)

GPIO.setup(in3,GPIO.OUT)
GPIO.setup(in4,GPIO.OUT)
GPIO.output(in3,GPIO.LOW)
GPIO.output(in4,GPIO.LOW)

## p=GPIO.PWM(en,100)
## p.start(0)
print("\n")
print("The default speed & direction of motor is LOW & Forward.....")
print("w-forward s-backward l-low m-medium h-high e-exit")
print("\n")

def ww():
    print("forward | destro e sinistro avanti")
    GPIO.output(in1,GPIO.LOW)
    GPIO.output(in2,GPIO.HIGH)
    GPIO.output(in3,GPIO.LOW)
    GPIO.output(in4,GPIO.HIGH)

def ss():
    print("backward | destro e sinistro indietro ")
    GPIO.output(in1,GPIO.HIGH)
    GPIO.output(in2,GPIO.LOW)
    GPIO.output(in3,GPIO.HIGH)
    GPIO.output(in4,GPIO.LOW)

def w(t):
    print("forward | destro e sinistro avanti")
    GPIO.output(in1,GPIO.LOW)
    GPIO.output(in2,GPIO.HIGH)
    GPIO.output(in3,GPIO.LOW)
    GPIO.output(in4,GPIO.HIGH)
    sleep(t)
    halt()

def s(t):
    print("backward | destro e sinistro indietro ")
    GPIO.output(in1,GPIO.HIGH)
    GPIO.output(in2,GPIO.LOW)
    GPIO.output(in3,GPIO.HIGH)
    GPIO.output(in4,GPIO.LOW)
    sleep(t)
    halt()

def l(t):
    print("l | destro avanti sinistro indietro ")
    GPIO.output(in1,GPIO.LOW)
    GPIO.output(in2,GPIO.HIGH)
    GPIO.output(in3,GPIO.HIGH)
    GPIO.output(in4,GPIO.LOW)
    sleep(t)
    halt()

def r(t):
    print("r | destro indietro sinistro avanti ")
    GPIO.output(in1,GPIO.HIGH)
    GPIO.output(in2,GPIO.LOW)
    GPIO.output(in3,GPIO.LOW)
    GPIO.output(in4,GPIO.HIGH)
    sleep(t)
    halt()

def halt():
    print("halt")
    GPIO.output(in1,GPIO.LOW)
    GPIO.output(in2,GPIO.LOW)
    GPIO.output(in3,GPIO.LOW)
    GPIO.output(in4,GPIO.LOW)


l(0.3)
r(0.3)

while(1):

    x=sys.stdin.read(1)   #raw_input() in python2

    print('Motors.py input', x)
    if x=='w':
        #### w(0.5)
        ww()
        x='z'

    elif x=='s':
        #### s(0.5)
        ss()
        x='z'

    elif x=='l':
        l(0.3)
        x='z'
    elif x=='a':
        l(0.3)
        x='z'
    elif x=='r':
        r(0.3)
        x='z'
    elif x=='d':
        r(0.3)
        x='z'

    elif x=='lold':
        print("left")
        GPIO.output(in1,GPIO.LOW)
        GPIO.output(in2,GPIO.HIGH)
        GPIO.output(in3,GPIO.LOW)
        GPIO.output(in4,GPIO.HIGH)
        sleep(0.5)
        halt()
        x='z'


    elif x=='rold':
        print("right")
        GPIO.output(in1,GPIO.LOW)
        GPIO.output(in2,GPIO.HIGH)
        GPIO.output(in3,GPIO.LOW)
        GPIO.output(in4,GPIO.HIGH)
        sleep(0.5)
        halt()
        x='z'

    elif x=='h':
        halt()
        x='z'

    elif x=='e':
        GPIO.cleanup()
        break

    #else:
        #print("<<<  wrong data  >>>")
        #print("please enter the defined data to continue.....")

File: unibo.basicrobot23/resources/robotNano/Motors.c
#include <iostream>
#include <wiringPi.h>
#include <fstream>
#include <cmath>
#include <stdio.h>

// #define TRUE 1   //alredy done in wiringPi
#define in1 9    /* 3 BCM */
#define in2 8    /* 2 BCM */

#define in3 12   /* 10 BCM */
#define in4 13   /*  9 BCM */

using namespace std;

int rotLeftTime  = 310;
int rotRightTime = 310;
int rotStepTime  = 58;


/*
    g++  Motors.c -l wiringPi -o  Motors
 */
void h(){
	digitalWrite(in2, LOW);
	digitalWrite(in1, LOW);
	digitalWrite(in3, LOW);
	digitalWrite(in4, LOW);
}

void w(){
    //print("forward | destro e sinistro avanti")
	digitalWrite(in1, LOW);
	digitalWrite(in2, HIGH);
	digitalWrite(in3, LOW);
	digitalWrite(in4, HIGH);
}
void s(){
    //print("backward | destro e sinistro indietro ")
	digitalWrite(in1, HIGH);
	digitalWrite(in2, LOW);
	digitalWrite(in3, HIGH);
	digitalWrite(in4, LOW);
}

void r(){
    //print("r | destro indietro sinistro avanti ")
	digitalWrite(in1, HIGH);
	digitalWrite(in2, LOW);
	digitalWrite(in3, LOW);
	digitalWrite(in4, HIGH);
	delay(rotRightTime);
	h();
}
void l(){
	digitalWrite(in1, LOW);
	digitalWrite(in2, HIGH);
	digitalWrite(in3, HIGH);
	digitalWrite(in4, LOW);
	delay(rotLeftTime);
	h();
}

//TODO
void aa(){
}
void dd(){
}
void z(){
	dd();
	delay(rotStepTime);
	h();
}
void x(){
	aa();
	delay(rotStepTime);
	h();
}

void configureRotationTime(){
}


void setup() {
	cout << "motorsC setUp STARTS" << endl;
	wiringPiSetup();
	pinMode(in2, OUTPUT);
	pinMode(in1, OUTPUT);
	pinMode(in3, OUTPUT);
	pinMode(in4, OUTPUT);
	h();
 	delay(30);
	cout << "motorsC setUp ENDS" << endl;
}

void remoteCmdExecutor(){
int input = 'h';
        input = getchar( );
        if( input != 10 ){
	        cout << input << endl;        
	        switch( input ){
	          case 99  : configureRotationTime(); break;  //c... | cl0.59 or cr0.59  or cx0.005 or cz0.005
	          case 119 : w(); break;  //w		 
	          case 115 : s(); break;  //s		 
	          case 97  : l(); break;  //a
	          case 122 : z(); break;  //z		 
	          case 120 : x(); break;  //x		 
	          case 100 : r(); break;  //d
	          case 104 : h();  break;  //h
	          case 114 : r();  break;  //r		  
	          case 108 : l(); break;    //l		 
	          case 10  :  break;    //lf
	          //case 102 : break;  //f
	          default  : h();
	        }
        }
}

int main(void) {
  	setup();
	while( 1 ) 	remoteCmdExecutor();
	//return 0;
}
